rust/noosphere-core/src/data/address.rs

```diff
@@ -65,13 +74,88 @@ pub struct LinkRecord(Jwt);
 impl LinkRecord {
     /// Parse the wrapped [Jwt] as a [Ucan] and looks for the referenced pointer
     /// to some data in IPFS (via a [Cid] in the `fct` field).
-    pub async fn dereference(&self) -> Option<Cid> {
-        let token = &self.0;
-        let ucan = match Ucan::try_from(token.to_string()) {
-            Ok(ucan) => ucan,
-            _ => return None,
+    pub fn dereference(&self) -> Option<Cid> {
+        match Ucan::try_from(self) {
+            Ok(ucan) => LinkRecord::link(&ucan),
+            _ => None,
+        }
+    }
+
+    /// Validates the [Ucan] token as a [LinkRecord], ensuring that
+    /// the sphere's owner authorized the publishing of a new
+    /// content address. Returns an `Err` if validation fails.
+    pub async fn validate<S: UcanJwtStore>(
+        token: &Ucan,
+        store: &S,
+        opt_did_parser: Option<&mut DidParser>,
```

Comment:
I've basically given up on including this in signatures. We should just instantiate `DidParser` at the point of use; it's just going to be removed when we update `rs-ucan` for the v0.10.0 UCAN spec.